src/pages/_app.js

```diff
@@ -22,9 +22,13 @@ if (typeof window !== 'undefined') {
 
 class BonusCalcApp extends App {
   render() {
-    const { Component, pageProps } = this.props
+    const { Component, pageProps, userDetails } = this.props
```

Comment:
Nitpick - just noticed that now we've destructured `userDetails` we can use it without accessing through `props` on lines 34 and 41.